### PR TITLE
ICMSLST-425 Show application contacts

### PR DESCRIPTION
**Pull request by kevinetienne**
* ICMSLST-862 Show agent dropdowns when creating applications

### CODE REVIEW

File: web/templates/web/domains/case/export/create.html [hunk 1]
```diff
@@ -49,4 +49,36 @@ <h3>{{ application_title }}</h3>
 {% block page_js %}
   {{ super() }}
   {{ form.media.js }}
+
+  <script type="text/javascript">
+    window.addEventListener('load', (event) => {
+      const exporters_with_agents = [
+        {%- for exporter in exporters_with_agents %}
+          '{{ exporter.pk }}'{% if not loop.last %},{% endif %}
+        {% endfor %}
```
Comment (MattHolmes123) [inline web/templates/web/domains/case/export/create.html:58]: I think we should separate the JS in to its own file and not mix jinja and JS.
e.g.
Set data in template:
https://github.com/uktrade/icms/blob/9c8143479399798a09a6c91decf94f2012a4a79a/web/templates/web/domains/case/import/opt/edit-compensating-products.html#L39
Parse it:
https://github.com/uktrade/icms/blob/9c8143479399798a09a6c91decf94f2012a4a79a/web/static/web/js/pages/opt-compensating-products.js#L20-L22

File: web/templates/web/domains/case/import/create.html [hunk 2]
```diff
@@ -37,9 +37,41 @@ <h3>{{ application_title }}</h3>
...
+          '{{ importer.pk }}'{% if not loop.last %},{% endif %}
+        {% endfor %}
+      ];
+
+      const importer = document.querySelector('#id_importer');
+      const agent = document.querySelector("#id_agent");
+      const agentOffice = document.querySelector("#id_agent_office");
+
+      const agentContainer = $(".row_id_agent");
+      const agentOfficeContainer = $(".row_id_agent_office");
```
Comment (MattHolmes123) [inline web/templates/web/domains/case/import/create.html:60]: Your choice but I'd honestly be happy never writing a line of jQuery again :smile: 

https://tobiasahlin.com/blog/move-from-jquery-to-vanilla-javascript/#hide-and-show

File: web/templates/web/domains/case/import/create.html [hunk 3]
```diff
@@ -37,9 +37,41 @@ <h3>{{ application_title }}</h3>
...
+
+      const triggerAgentDropdowns = () => {
+        if (importers_with_agents.includes(importer.value)) {
+          agentContainer.show();
+          agentOfficeContainer.show();
+        } else {
+          agentContainer.hide();
+            agentOfficeContainer.hide();
+        }
```
Comment (MattHolmes123) [inline web/templates/web/domains/case/import/create.html:69]:
```
let displayStyle = importers_with_agents.includes(importer.value) ? "block": "none"
agentContainer.style.display = displayStyle
agentOfficeContainer.style.display = displayStyle
```

File: web/templates/web/domains/case/import/create.html [hunk 4]
```diff
@@ -37,9 +37,41 @@ <h3>{{ application_title }}</h3>
...
+      };
+
+      triggerAgentDropdowns();
+
+      $("#id_importer").on("change", (e) => triggerAgentDropdowns());
```
Comment (MattHolmes123) [inline web/templates/web/domains/case/import/create.html:74]: `importer.addEventListener("change", (e) => {})`

File: web/templates/web/domains/case/export/create.html [hunk 5]
```diff
@@ -49,4 +49,36 @@ <h3>{{ application_title }}</h3>
...
+      ];
+
+      const exporter = document.querySelector('#id_exporter');
+      const agent = document.querySelector("#id_agent");
+      const agentOffice = document.querySelector("#id_agent_office");
+
+      const agentContainer = $(".row_id_agent");
```
Comment (MattHolmes123) [inline web/templates/web/domains/case/export/create.html:65]: I added several comments to the next file thinking it was this one.  
But all comments apply to both blocks of JS.
I think they should both be in a JS file.

**Review by MattHolmes123 — CHANGES_REQUESTED** [review]

File: web/templates/web/domains/case/export/create.html (hunk 1 above)
Comment (kevinetienne) [reply web/templates/web/domains/case/export/create.html]: updated in https://github.com/uktrade/icms/pull/503/commits/529ae8d1cc3af570edabf59acb2350874ad8e5c7 thanks

File: web/templates/web/domains/case/import/create.html (hunk 3 above)
Comment (kevinetienne) [reply web/templates/web/domains/case/import/create.html]: updated in https://github.com/uktrade/icms/pull/503/commits/529ae8d1cc3af570edabf59acb2350874ad8e5c7

File: web/templates/web/domains/case/import/create.html (hunk 2 above)
Comment (kevinetienne) [reply web/templates/web/domains/case/import/create.html]: updated in https://github.com/uktrade/icms/pull/503/commits/529ae8d1cc3af570edabf59acb2350874ad8e5c7

File: web/templates/web/domains/case/export/create.html (hunk 5 above)
Comment (kevinetienne) [reply web/templates/web/domains/case/export/create.html]: updated in https://github.com/uktrade/icms/pull/503/commits/529ae8d1cc3af570edabf59acb2350874ad8e5c7

File: web/templates/web/domains/case/import/create.html (hunk 4 above)
Comment (kevinetienne) [reply web/templates/web/domains/case/import/create.html]: I couldn't use plain js to add the event listener. As it wasn't triggered. I guess select2 update the `select` behind the scene and doesn't trigger the `change` event.

Comment (MattHolmes123) [reply web/templates/web/domains/case/import/create.html]: Yer fair enough :+1:

**Review by MattHolmes123 — APPROVED** [review]
Mentioned to Kevin a few minor things, approving while he changes them.

**Pull request merged**